### PR TITLE
go1.23 support

### PR DESCRIPTION
**Pull request by ldez**
This PR is to evaluate and prepare golangci-lint to [go1.23](https://tip.golang.org/doc/go1.23)

This PR will evolve during the beta and RC phases of [go1.23](https://tip.golang.org/doc/go1.23).

**Questions and problems related to go1.23 should be reported in the issue #4837.**

---

`golangci-lint` is a free and open-source project built by volunteers.

If you value it, consider supporting us, we appreciate it! :heart:

[Open Collective backers and sponsors](https://opencollective.com/golangci-lint)

[GitHub Sponsors](https://github.com/sponsors/golangci)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -127,7 +127,7 @@ require (
 	golang.org/x/exp v0.0.0-20240103183307-be819d1f06fc
 	golang.org/x/tools v0.24.0
 	gopkg.in/yaml.v3 v3.0.1
-	honnef.co/go/tools v0.4.7
+	honnef.co/go/tools v0.5.0-rc.1
```
Comment (ryancurrah) [inline go.mod:130]: Note this is a pre-release of static check.

Comment (ldez) [reply go.mod]: staticcheck just release v0.5.0

Comment (ryancurrah) [reply go.mod]: Should you update to v0.5.0 for this release?

Comment (ryancurrah) [reply go.mod]: nvm 👍

Comment (ldez) [reply go.mod]: yes it's a requirement to support go1.23

**Review by ryancurrah — APPROVED** [review]

**Pull request merged**